dnf/persistor.py

```diff
@@ -84,12 +84,20 @@ def _last_makecache_path(self):
         return os.path.join(self.cachedir, "last_makecache")
 
     def get_expired_repos(self):
-        self._check_json_db(self.db_path)
-        return set(self._get_json_db(self.db_path))
+        try:
+            self._check_json_db(self.db_path)
+            return set(self._get_json_db(self.db_path))
+        except OSError as e:
+            logger.info(_("Failed to load expired repos cache: %s"), e)
```

Comment:
AFAICS `logger.info` logs to `stdout`, this should certainly go to `stderr` and I think it should be a warning anyway? (`logger.warning` should go to `stderr`)

Also, can you include the path to the file in the message? That makes it much easier for the user to check what's the problem (without it, they are left to guess where to look).

Applies to the equivalent message in `save(self)` as well.

Reply:
> AFAICS `logger.info` logs to `stdout`, this should certainly go to `stderr` and I think it should be a warning anyway? (`logger.warning` should go to `stderr`)

I was being consistent with messages of makecache failures, but warning does make more sense. And I would change the makecache messages, too, if you agree.

> Also, can you include the path to the file in the message? That makes it much easier for the user to check what's the problem (without it, they are left to guess where to look).

In cases I tried, the exception message itself contains the path to the file, so it would be there twice. But there probably might be other cases where it would be useful, so I can add it.

Reply:
I made the changes and also changed the tests accordingly.

Reply:
Ok, thanks. I didn't realize `OSError` contains the file name, in C/C++ it's not part of the error and I was thinking it's the same here... If the file name is there, I think we can assume it should be there for all cases of `OSError` where it's available and we don't need to mention it twice (it makes the error message ugly). Sorry about the confusion.

Reply:
All right then.

Btw, I just noticed the tests are not properly referenced here, so: https://github.com/rpm-software-management/ci-dnf-stack/pull/865